components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/MetricHandler.java

```diff
@@ -0,0 +1,582 @@
+    private void incrementApiErrorCount(String name, String apiInvocationUrl) {
+        Map<String, String[]> metricCountMap = new HashMap();
+        metricCountMap.put(MetricConstants.API_REQUEST_COUNT_ERROR_TOTAL, new String[]{name,
+                SynapseConstants.FAIL_SAFE_MODE_API, apiInvocationUrl});
+
+        // Invoke the incrementCount() method in the dynamically loaded MetricHandler instance.
+        Class[] incrementApiErrorCountMethodArgs = {String.class, Map.class};
+        invokeIncrementCount(INCREMENT_COUNT, incrementApiErrorCountMethodArgs,
+                MetricConstants.API_REQUEST_COUNT_ERROR_TOTAL, metricCountMap);
+    }
+
+    /**
+     * Increment the error request count received by an inbound endpoint.
+     *
+     * @param name The metric name
+     */
+    private void incrementInboundEndpointErrorCount(String name) {
+        Map<String, String[]> metricCountMap = new HashMap();
+        metricCountMap.put(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_ERROR_TOTAL, new String[]{name,
+                MetricConstants.INBOUND_ENDPOINT});
+
+        // Invoke the incrementCount() method in the dynamically loaded MetricHandler instance.
+        Class[] incrementInboundEndpointErrorCountMethodArgs = {String.class, Map.class};
+        invokeIncrementCount(INCREMENT_COUNT, incrementInboundEndpointErrorCountMethodArgs,
+                MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_ERROR_TOTAL, metricCountMap);
+    }
+
+    /**
+     * Get the api name.
+     *
+     * @param contextPath The api context path
+     * @param synCtx The Synapse Message Context
+     * @return String The api name
+     */
+    private String getApiName(String contextPath, MessageContext synCtx) {
+        String apiName = null;
+        for (API api : synCtx.getEnvironment().getSynapseConfiguration().getAPIs()) {
+            if (restUtils.getRESTApiName(contextPath, api.getContext())) {
+                apiName = api.getName();
+                if (api.getVersionStrategy().getVersion() != null && !"".equals(api.getVersionStrategy().
+                        getVersion())) {
+                    apiName = apiName + ":v" + api.getVersionStrategy().getVersion();
+                }
+                synCtx.setProperty(MetricConstants.IS_ALREADY_PROCESSED_REST_API, true);
+                synCtx.setProperty(MetricConstants.PROCESSED_API, api);
+            }
+        }
+        return apiName;
+    }
+
+    /**
+     * Invoke the invokeIncrementCount() method.
+     *
+     * @param methodName                     Name of the method needs to be invoked
+     * @param invokeIncrementCountMethodArgs Array of Class objects that identify the method's formal parameter types
+     * @param metricName                     Metric Name
+     * @param metricValue                    Map containing the <metricname,labels> as key,value pairs
+     */
+    private Object invokeIncrementCount(String methodName, Class[] invokeIncrementCountMethodArgs,
```

Comment:
All these methods need to change. You dont need to use reflextion. you can get an instance of the class

Reply:
Class clazz = Class.forName(YOUR_CLASS_NAME);
MetricsReporter metricsReporter = (MetricsReporter) clazz.newInstance();

Reply:
Fixed